util/HTTPServer.js

```diff
@@ -76,10 +76,17 @@ function HTTPServer(app) {
         server.use(passport.initialize());
         server.use((req, res, next) => {
             function authUser(user) {
+                if (user && user.deactivated) {
+                    user.deactivated = false;
+                    user.deletionDate = null;
+                    user.save();
```

Comment:
has this been fully tested? I'm skeptical because `user.save()` is async

Reply:
Ideally this should be awaited, but it’s been tested and does work. There’s a few places in this PR where I don’t await user.save; I’ll fix them in a bit.